### PR TITLE
Feature/sdk-708_update_cdr_config_APIs

### PR DESCRIPTION
**Pull request by NeilCossorFrollo**
https://frollo.atlassian.net/browse/SDK-708
Links to API docs in that ticket.
Note there are a number of mandatory new fields, including one defined in 2.15 which was missing. To get database migration tests working, needed to add defaults in xcdatamodel which will be stamped over on config/cdr refresh.
Unit tests for a bunch of decode failures implemented using simple string substitution. Not perfect where same "key" is in different levels of nested JSON subfields e.g. "id" or "name"
Note at time of this PR, the backend ticket is NOT in place, so purely relies on JSON manually generated from blend of real response data, and documented sample data in beta API docs.

Checklist
Use the ✅ emoji for when it's been done; ⛔ when it still needs work; strikethrough when an item of the checklist does not apply to the changes.

- Is the destination branch correct & intended? ✅ 
- Is the correct JIRA ticket referenced? ✅ 
- Is the correct API docs/Confluence Page referenced? ✅ 
- Have all the AC's in the ticket passed? ✅ 
- Have you followed proper naming conventions in the code? ✅ 
- Has the code been properly covered with unit tests? ✅ (not 100% for failure cases, better than was there)
- Have you regenerated the mock DBs, if the datamodel was modified ✅ 
~~Have you added migration tests (if applicable)?~~ N/A (same 1.7.8)
- Have the unit tests passed locally (if applicable)? ✅ 
- Have you addressed the code warnings if any? ✅ 
- Have you tested the API and made sure the documentation is up-to-date? ✅ Given API docs are beta, not implemented yet
- Have you added documentation for all public methods & types ✅

### CODE REVIEW

File: Tests/Resources/API/CDR/get_cdrConfiguration_2RP.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
+{
+    "support_email": "support@acme.com",
+    "adr_id": "ADBK0003",
+    "adr_name": "ACME",
+    "cdr_policy_url": "https://example.com",
+    "display_name": "Frollo",
+    "external_id": "frollo-default",
+    "id": "2",
```
Comment (dipeshfrollo) [inline Tests/Resources/API/CDR/get_cdrConfiguration_2RP.json:8]: Isnt ID supposed to be an Int64?

File: Sources/FrolloSDK/Model/API/CDR/APICDRConfigurationResponse.swift [hunk 2]
```diff
@@ -32,17 +45,31 @@ struct APICDRConfigurationResponse: Codable {
     }
     
     enum CodingKeys: String, CodingKey {
-        case supportEmail = "support_email"
         case adrID = "adr_id"
         case adrName = "adr_name"
-        case sharingDurations = "sharing_durations"
+        case cdrPolicyURL = "cdr_policy_url"
```
Comment (dipeshfrollo) [inline Sources/FrolloSDK/Model/API/CDR/APICDRConfigurationResponse.swift:50]: I think we will also need a required field "id"

**Review by dipeshfrollo — CHANGES_REQUESTED** [review]

File: Tests/Resources/API/CDR/get_cdrConfiguration_2RP.json (hunk 1 above)
Comment (NeilCossorFrollo) [reply Tests/Resources/API/CDR/get_cdrConfiguration_2RP.json:8]: With some of the IDs as string and some as integer (and not knowing who is allocating a specific ID), I missed that the 2.17 docs say this is Integer, so updating to Int64.

File: Sources/FrolloSDK/Model/API/CDR/APICDRConfigurationResponse.swift (hunk 2 above)
Comment (NeilCossorFrollo) [reply Sources/FrolloSDK/Model/API/CDR/APICDRConfigurationResponse.swift:50]: it is there, with coding key from id to configID

Comment (hailefrollo) [reply Sources/FrolloSDK/Model/API/CDR/APICDRConfigurationResponse.swift:50]: I notice the `id` now BE return the number, not string. 🤷‍♂️

**Comment by NeilCossorFrollo** [comment]
Updated with different endpoint, so get now required externalID as parameter, which ripples up the chain thru Link and app (externalID generated from tenant)

File: Sources/FrolloSDK/Aggregation/Aggregation.swift [hunk 3]
```diff
@@ -466,8 +466,9 @@ public class Aggregation: CachedObjects, ResponseHandler {
      - parameters:
         - completion: Optional completion handler with optional error if the request fails
      */
-    public func refreshCDRConfiguration(completion: FrolloSDKCompletionHandler? = nil) {
-        service.fetchCDRConfiguration { result in
+    public func refreshCDRConfiguration(externalID: String,
```
Comment (dipeshfrollo) [inline Sources/FrolloSDK/Aggregation/Aggregation.swift:469]: Can we add new parameter in documentation please?

File: Tests/Database/DatabaseMigrationTests.swift [hunk 4]
```diff
@@ -120,18 +129,26 @@ class DatabaseMigrationTests: XCTestCase {
...
     }
     
     // MARK: - Test Data
     
     // Uncomment to generate test data for each data model. Do this whenever adding a new Core Data model version
-//    func testGenerateTestCoreDataDatabases() {
-//        generateCoreDataModelTestDatabases()
-//        generateFakeCoreDataModelTestDatabase()
-//    }
+    func testGenerateTestCoreDataDatabases() {
```
Comment (dipeshfrollo) [inline Tests/Database/DatabaseMigrationTests.swift:142]: Lets not forget to comment this

**Review by dipeshfrollo — APPROVED** [review]
Looks good. Just 2 minor comments.

File: Tests/Database/DatabaseMigrationTests.swift (hunk 4 above)
Comment (NeilCossorFrollo) [reply Tests/Database/DatabaseMigrationTests.swift]: fixed in last commit

File: Sources/FrolloSDK/Aggregation/Aggregation.swift (hunk 3 above)
Comment (NeilCossorFrollo) [reply Sources/FrolloSDK/Aggregation/Aggregation.swift:470]: added comment for new param

**Pull request merged**